consensus/beacon/consensus.go

```diff
@@ -382,8 +383,50 @@ func (beacon *Beacon) FinalizeAndAssemble(chain consensus.ChainHeaderReader, hea
 	// Assign the final state root to header.
 	header.Root = state.IntermediateRoot(true)
 
+	var (
+		p    *verkle.VerkleProof
+		k    verkle.StateDiff
+		keys = state.Witness().Keys()
+	)
+	if chain.Config().IsVerkle(header.Number, header.Time) {
```

Comment:
This is the extra part: if we are past the fork, generate the proof to go with it